src/arviz_plots/plots/lm_plot.py

```diff
@@ -0,0 +1,324 @@
+"""lm plot code."""
+from collections.abc import Mapping, Sequence
+from copy import copy
+from importlib import import_module
+from typing import Any, Literal
+
+import numpy as np
+import xarray as xr
+from arviz_base import rcParams
+from scipy.interpolate import griddata
+from scipy.signal import savgol_filter
+
+from arviz_plots.plot_collection import PlotCollection
+from arviz_plots.plots.utils import (
+    filter_aes,
+    get_contrast_colors,
+    process_group_variables_coords,
+    set_wrap_layout,
+)
+from arviz_plots.visuals import fill_between_y, labelled_x, labelled_y, line_xy, scatter_xy
+
+
+def plot_lm(
+    dt,
+    target_data=None,
+    independent_data=None,
+    prediction_data=None,
+    ci_mean_independent_data=None,
```

Comment:
we should discuss these argument names tomorrow

Reply:
target_data -> y 
independent_data -> x
prediction_data -> y_pred
ci_mean_independent_data -> x_pred